index.js

```diff
@@ -1,12 +1,8 @@
 import {BossRoomBot, CharInfo} from "./bossroom";
-import {RGValidator} from "./rg";
-
-
-let charType = Math.round(Math.random() * 1000000) % 4;
+import {MathFunctions, RGValidator} from "./rg";
 
 export function configureBot(characterType) {
-  console.log(`Unity bot configureBot function called, charType: ${charType} - characterType: ${characterType}`);
-  charType = CharInfo.type.indexOf(characterType);
+  console.log(`Unity bot configureBot function called, charType: ${getCharacterType()} - characterType: ${characterType}`);
```

Comment:
this breaks the server random assignment case , please revert this change

Reply:
For the co-op bot I wanted to only use the archer, and then was going to have the ability bot keep this functionality. I was a little confused by this code actually, if the getCharacterType function is configuring the bot type, what is this setting exactly?

Reply:
charType field is randomly assigned when the bot code instantiates

so getCharacterType() will return that value until the handshake completes (unless you override it)

once the handshake completes

 ... if you supplied a value for characterType, you will get that same value back and charType will be set to that same value

 ... if you didn't supply a value for characterType, you get some random value back which will then be set as your charType

Reply:
Got it, thanks!